src/mono/mono/metadata/assembly.c

```diff
@@ -1769,6 +1773,50 @@ netcore_load_reference (MonoAssemblyName *aname, MonoAssemblyLoadContext *alc, M
 	return reference;
 }
 
+MonoImage *
+netcore_open_from_bundle (MonoAssemblyLoadContext *alc, const char *filename, MonoImageOpenStatus *status, gboolean refonly, const char* culture)
+{
+	int i;
+	char *name;
+	MonoImage *image = NULL;
+	gboolean is_satellite = culture && culture[0] != 0;
+
+	if (!is_satellite)
+	{
+		if (!bundles)
+			return NULL;
+
+		name = g_path_get_basename (filename);
+		for (i = 0; !image && bundles [i]; ++i) {
+			if (strcmp (bundles [i]->name, name) == 0) {
+				image = mono_image_open_from_data_internal (alc, (char*)bundles [i]->data, bundles [i]->size, FALSE, status, refonly, FALSE, name, NULL);
+				break;
+			}
+		}
+	}
+	else
+	{
+		if (!satellite_bundles)
+			return NULL;
+		
+		name = strdup(filename);
+		for (i = 0; !image && satellite_bundles [i]; ++i) {
+			if (strcmp (satellite_bundles [i]->culture, culture) == 0 && strcmp (satellite_bundles [i]->name, name) == 0) {
```

Comment:
You should probably compare the name first since it's less likely to match (and then you avoid the second comparison). Not that it matters that much, but hey!

Reply:
The reason why I decided to do the culture comparison first was because it was a shorter string comparison, I was doubtful of which one to do first. If you think it is better to do the name comparison first, then I'll do that one.